accounts-db/src/tiered_storage/hot.rs

```diff
@@ -540,8 +542,11 @@ impl HotStorageWriter {
         accounts: &StorableAccountsWithHashesAndWriteVersions<'a, 'b, T, U, V>,
         skip: usize,
     ) -> TieredStorageResult<()> {
+        let zero_lamport_account_owner: Pubkey = Pubkey::default();
```

Comment:
I actually tried this first, but Rust complains that I cannot do this:

```
   Compiling solana-accounts-db v1.18.0 (/Users/keelar/solana/yhchiang-sol/accounts-db)
error[E0015]: cannot call non-const fn `<solana_sdk::pubkey::Pubkey as std::default::Default>::default` in constants
   --> accounts-db/src/tiered_storage/hot.rs:545:40
    |
545 |         const DEFAULT_PUBKEY: Pubkey = Pubkey::default();
    |                                        ^^^^^^^^^^^^^^^^^
    |
    = note: calls in constants are limited to constant functions, tuple structs and tuple variants
```

Let me try the lazy_init again (I think I might have tried that before as well, forgot what was complained).

Reply:
> This basically guarantees every hot storage file will have one additional owner that represents "no owner". We could also optimize this away by storing some flag

I think one way to achieve this is to have one special value of OwnerOffset indicating no owner (probably OwnerOffset(u32::MAX) I think).  This is probably a better way to handle it.  Let me try to handle it.

Reply:
Damn.  I need to pick some value other than u32::MAX because hot-storage stores owners offset using only 29 bits.

```
 struct HotMetaPackedFields {
     /// A hot account entry consists of the following elements:
     ///
     /// * HotAccountMeta
     /// * [u8] account data
     /// * 0-7 bytes padding
     /// * optional fields
     ///
     /// The following field records the number of padding bytes used
     /// in its hot account entry.
     padding: B3,
     /// The index to the owner of a hot account inside an AccountsFile.
     owner_offset: B29,
 }
```

In addition, a MAX value like this also becomes a special case in the boundary check.  Will probably do it in a separate PR.

Reply:
Use const OWNER_NO_OWNER defined in owners.rs while keeping the OwnerOffset logic the same.  We can continue the discussion and I will create a separate PR for how to avoid saving extra owner per file.

Reply:
My thought on this is that: I am not sure whether we might change the bits later when we already decided a const limit (for instance, 28 should be more than enough, but I am not sure whether we will squeeze the bit further for other purpose and change the limit to 27 bits).

If we want a const to represent the owner offset for no owner that we will never change, then 0 could be a good one?  But as a downside, we also need to properly do `-1` when reading any entry.